### PR TITLE
Update to latest api

### PR DESCRIPTION
**Pull request by molpopgen**
1. Update the WF forward-time tutorial to the current API.
2. Fix the recommended recombination rate when starting a simulation using a tree seuqence from msprime.  @petrelharp @jeromekelleher -- this puzzles me.  When working on the paper, I did a lot of testing suggesting that the old recommendation was correct, but that is no longer the case.  It is possible that I was wrong before....  thoughts?
3. Require python3 for building.

### CODE REVIEW

File: docs/wfforward.md [hunk 1]
```diff
@@ -1,12 +1,14 @@
 
 # Tracking the genealogy during forward simulation
 
-This tutorial will cover several use cases of `tskit` for forward simulations.  We start with a simple Wright-Fisher simulation with no selection and no recombination, and gradually increase the complexity of our examples until we are recording mutations, ancient samples, and associated meta-data.
-
-We will record all data using `tskit` machinery contained in `msprime`.  We will show how to periodically "garbage collect", which means apply the tree sequence simplification algorithm, at regular intervals, which keeps RAM use controlled.
+This tutorial will cover several use cases of `tskit` for forward simulations.  We start with a simple Wright-Fisher simulation with no selection and no recomination, and gradually increase the complexity of our examples until we are recording mutations, ancient samples, and associated meta-data.
```
Comment (jeromekelleher) [inline docs/wfforward.md]: "recomination"

File: docs/wfforward.md [hunk 2]
```diff
@@ -90,20 +92,20 @@ def wf1(N, ngens):
...
     for i in range(2*N):
-        nodes.add_row(time=0, flags=msprime.NODE_IS_SAMPLE)
+        tc.nodes.add_row(time=0, flags=msprime.NODE_IS_SAMPLE)
     
-    next_offspring_index = len(nodes)
+    next_offspring_index = len(tc.nodes)
     first_parental_index = 0
     for gen in range(1,ngens+1):
-        assert next_offspring_index == len(nodes)
-        assert first_parental_index == len(nodes) - 2*N
```
Comment (jeromekelleher) [inline docs/wfforward.md]: Making ``assert`` look like a function isn't terribly idiomatic I think, but it's a personal style choice (so up to you).

File: docs/wfforward.md [hunk 3]
```diff
@@ -828,17 +825,25 @@ Let's take a look at how many roots are present in each marginal tree that came
...
     nroots.append(i.num_roots)
+print(np.unique(nroots))
 p = sns.distplot(nroots, kde=False)
 p.set(xlabel="Number of roots",ylabel="Number");
 ```
 
+    [1 2]
+
+
+    /usr/lib/python3/dist-packages/matplotlib/axes/_axes.py:6462: UserWarning: The 'normed' kwarg is deprecated, and has been replaced by the 'density' kwarg.
```
Comment (jeromekelleher) [inline docs/wfforward.md]: Do we want this in the output?

**Comment by jeromekelleher** [comment]
> Fix the recommended recombination rate when starting a simulation using a tree seuqence from msprime. @petrelharp @jeromekelleher -- this puzzles me. When working on the paper, I did a lot of testing suggesting that the old recommendation was correct, but that is no longer the case. It is possible that I was wrong before.... thoughts?

I missed this bit @molpopgen --- where are we doing this?

**Comment by molpopgen** [comment]
> I missed this bit @molpopgen --- where are we doing this?

[here](https://github.com/molpopgen/fwdpy11_arg_example/blob/master/fwdpy11_arg_example/evolve_arg.py#L57)

I distinctly remember testing this, and there was discussion during writing the paper.  However, my tests involved starting with msprime output and then simulating for 10N generations, which I think means any error is relatively small.  More recent tests in fwdpy11 show this scaling to be wrong.

What are you doing in msprime's "coalesce back", and are there tests on the distribution of S?

File: docs/wfforward.md (hunk 3 above)
Comment (molpopgen) [reply docs/wfforward.md]: No--any ideas how to suppress it?

Comment (jeromekelleher) [reply docs/wfforward.md]: Upgrade seaborn, something like that?

Comment (molpopgen) [reply docs/wfforward.md]: I just suppressed warnings for those plotting calls.

**Pull request closed without merging**